### PR TITLE
Location was removed from abstract-leveldown

### PR DESCRIPTION
**Pull request by ralphtheninja**
* handle location in LevelDOWN constructor
* pass a factory function to abstract tests
* use tempy and remove cleanup code

### CODE REVIEW

**Comment by ralphtheninja** [comment]
@vweevers How do you see us using this branch? Will it be a long lasting branch? It will always target `abstract-leveldown#master` so I guess we can cherry-pick from this onto master branch once a new `abstract-leveldown` has been released?

**Comment by vweevers** [comment]
>  It will always target abstract-leveldown#master so I guess we can cherry-pick from this onto master branch once a new abstract-leveldown has been released?

Or rebase to remove the first commit, then next can be merged into master. I don't know, we'll see, also depends on what other stuff happens on master in the mean time.

Main reason I suggested this approach is to have a way to canary test `abstract-leveldown` and keep discussion on unforeseen issues focused (by having small PRs). I wasn't thinking too much about what happens after.

**Comment by ralphtheninja** [comment]
> Main reason I suggested this approach is to have a way to canary test abstract-leveldown and keep discussion on unforeseen issues focused (by having small PRs). I wasn't thinking too much about what happens after.

It's a good idea. We should have this on `memdown`, `level-js` and `rocksdb` as well.

File: leveldown.js [hunk 1]
```diff
@@ -9,7 +9,9 @@ function LevelDOWN (location) {
     return new LevelDOWN(location)
   }
 
-  AbstractLevelDOWN.call(this, location)
+  AbstractLevelDOWN.call(this)
+
+  this.location = location
```
Comment (vweevers) [inline leveldown.js]: We need a type check here (https://github.com/Level/abstract-leveldown/blob/8ca1de858a6c98d9de4eb1063aba670d57df7afa/abstract-leveldown.js#L14-L16)

File: test/leveldown-test.js [hunk 2]
```diff
@@ -1,5 +1,15 @@
...
 abstract.args(testCommon.factory, test)
+
+test('test database creation non-string location throws', function (t) {
+  t.throws(
+    leveldown.bind(null, {}),
+    /constructor requires a location string argument/,
+    'non-string location leveldown() throws'
+  )
+  t.end()
+})
```
Comment (ralphtheninja) [inline test/leveldown-test.js]: This build will fail since master has changed.

Comment (vweevers) [reply test/leveldown-test.js]: Perhaps we should target commit refs?

Comment (ralphtheninja) [reply test/leveldown-test.js]: Funny thing, the tests doesn't fail.

Comment (ralphtheninja) [reply test/leveldown-test.js]: Just goes from

screenshot from 2018-07-13 11-19-58

to

screenshot from 2018-07-13 11-19-21

Comment (ralphtheninja) [reply test/leveldown-test.js]: > Perhaps we should target commit refs?

I'll target a commit ref in next PR.

**Pull request merged**